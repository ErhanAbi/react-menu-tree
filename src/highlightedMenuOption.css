.highlight {
  font-weight: bold;
  color: tomato;
}
