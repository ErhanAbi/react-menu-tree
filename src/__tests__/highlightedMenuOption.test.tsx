import React from "react";
import renderer from "react-test-renderer";

import { HighlightedMenuOption } from "../highlightedMenuOption.";

describe(HighlightedMenuOption, () => {
  it("renders text without highlights if no match found", () => {
    const wrapper = renderer
      .create(
        <HighlightedMenuOption
          text="Some Text"
          searchedValue="lazy fox nowhere to be found"
          highlightClass="highlight"
        />
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders text with highlight if text was found", () => {
    const wrapper = renderer
      .create(
        <HighlightedMenuOption
          text="Lazy Fox Hiding"
          searchedValue="lazy fox"
          highlightClass="highlight"
        />
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
