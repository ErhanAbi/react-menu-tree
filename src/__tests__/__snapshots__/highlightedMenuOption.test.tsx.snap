// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightedMenuOption renders text with highlight if text was found 1`] = `
Array [
  "",
  <span
    className="highlight undefined"
  >
    Lazy Fox
  </span>,
  " Hiding",
]
`;

exports[`HighlightedMenuOption renders text without highlights if no match found 1`] = `"Some Text"`;
